src/SelfServe/SqlX/SqlX.tsx

```diff
@@ -50,30 +66,88 @@ export default class SqlX extends SelfServeBaseClass {
   };
 
   public onSave = async (currentValues: Map<string, SmartUiInput>): Promise<SelfServeNotification> => {
-    validate(currentValues);
-    // TODO: add pre processing logic before calling the updateDedicatedGatewayProvisioning() RP call.
-    throw new Error(`onSave not implemented. No. of properties to save: ${currentValues.size}`);
+    const response = await getCurrentProvisioningState();
+
+    // null implies the resource has not been provisioned.
+    if (response.status !== undefined && response.status !== ResourceStatus.Running.toString()) {
+      switch (response.status) {
+        case ResourceStatus.Creating.toString():
+          return { message: "CreateMessage", type: SelfServeNotificationType.error };
+        case ResourceStatus.Updating.toString():
+          return { message: "UpdateMessage", type: SelfServeNotificationType.error };
+        case ResourceStatus.Deleting.toString():
+          return { message: "DeleteMessage", type: SelfServeNotificationType.error };
+        default:
+          return { message: "CannotSave", type: SelfServeNotificationType.error };
+      }
+    }
+
+    const enableDedicatedGateway = currentValues.get("enableDedicatedGateway")?.value as boolean;
+
+    if (response.status !== undefined) {
+      if (!enableDedicatedGateway) {
+        try {
+          await deleteDedicatedGatewayResource();
+          return { message: "DedicatedGateway resource will be deleted.", type: SelfServeNotificationType.info };
+        } catch (e) {
+          return {
+            message: "Deleting Dedicated Gateway resource failed. DedicatedGateway will not be deleted.",
+            type: SelfServeNotificationType.error,
+          };
+        }
+      } else {
+        // Check for scaling up/down/in/out
```

Comment:
what logic goes here?

Reply:
Scaling up logic, the RP team is still working on it. Currently Updates with changed number of Instances does not work.